### PR TITLE
devops: refactor GitHub Actions workflows

### PR DESCRIPTION
**Pull request by mxschmitt**
https://github.com/microsoft/playwright-internal/issues/203

### CODE REVIEW

File: .github/workflows/tests_primary.yml [hunk 1]
```diff
@@ -49,33 +49,16 @@ jobs:
...
-    - run: ./utils/upload_flakiness_dashboard.sh ./test-results/report.json
-      if: ${{ !cancelled() }}
-      shell: bash
-    - name: Upload blob report
-      if: ${{ !cancelled() }}
-      uses: ./.github/actions/upload-blob-report
-      with:
-        report_dir: blob-report
-        job_name: ${{ env.PWTEST_BOT_NAME }}
+        is-linux: true
```
Comment (dgozman) [inline .github/workflows/tests_primary.yml:55]: I'd say we should pass `os` if we can.

File: .github/workflows/tests_primary.yml [hunk 2]
```diff
@@ -49,33 +49,16 @@ jobs:
...
+        validate-coverage: ${{ matrix.browser }}
+        browser-to-install: ${{ matrix.browser }}
```
Comment (dgozman) [inline .github/workflows/tests_primary.yml:57]:
```suggestion
        browsers-to-install: ${{ matrix.browser }}
```

File: .github/workflows/tests_primary.yml [hunk 3]
```diff
@@ -49,33 +49,16 @@ jobs:
...
+        command: npm run test -- --project=${{ matrix.browser }}-*
```
Comment (dgozman) [inline .github/workflows/tests_primary.yml:58]: - We'll also need env for trace/video/etc bots.
- I'd also make "PWTEST_BOT_NAME" an argument.

File: .github/workflows/tests_primary.yml (hunk 1 above)
Comment (mxschmitt) [reply .github/workflows/tests_primary.yml]: I got rid of it, we can make the condition inside bash.

File: .github/actions/run-test/action.yml [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    description: 'Browser to install. Default is all browsers.'
+    required: false
+    default: ''
+  check-coverage:
+    description: 'Validate coverage. Default is false. Pass the browser name to validate coverage.'
+    required: false
+  bot-name:
+    description: 'Bot name'
+    required: true
+  flakiness-client-id:
```
[the remark this answers is not in the document]
Comment (mxschmitt) [reply .github/actions/run-test/action.yml:18]: Unfortunately not. Composite Actions don't have access to secrets/vars as per [here](https://github.com/orgs/community/discussions/49689).

**Review by yury-s — APPROVED** [review]

**Pull request merged**